File: leetcode/1009_Complement_of_Base_10_Integer.py
# URL: https://leetcode.com/problems/complement-of-base-10-integer/?envType=daily-question&envId=2026-03-11
#
# Problem: 1009. Complement of Base 10 Integer
#

# Solution:
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        binary = bin(n)[2:]
        ans = ""
        for i in binary:
            if i == "1":
                ans += "0"
            else:
                ans += "1"
        return int(ans,2)

File: leetcode/1009_Complement_of_Base_10_Integer.java
// URL: https://leetcode.com/problems/complement-of-base-10-integer/submissions/1944567351/?envType=daily-question&envId=2026-03-11
//
// Problem: 1009. Complement of Base 10 Integer
//

// Solution:
class Solution {
    public int bitwiseComplement(int n) {
        String s = Integer.toBinaryString(n);
        String sb = "";
        for(char i : s.toCharArray()){
            sb += i == '0' ? '1' : '0';
        }
        System.out.println(sb);
        return Integer.parseInt(sb,2);
    }
}